api/src/main/java/au/org/aodn/nrmn/restapi/service/SpreadSheetService.java

```diff
@@ -42,150 +38,47 @@
-                .mapToObj(index -> {
-                    val row = dataSheet.getSheet().getRow(index);
-                    val stagedRow = new StagedRow();
-                    stagedRow.setDiver(_getCellValue(row.getCell(headerMap.get("Diver")), eval, fmt));
-                    stagedRow.setBuddy(_getCellValue(row.getCell(headerMap.get("Buddy")), eval, fmt));
-                    stagedRow.setSiteCode(_getCellValue(row.getCell(headerMap.get("Site No.")), eval, fmt));
-                    stagedRow.setSiteName(_getCellValue(row.getCell(headerMap.get("Site Name")), eval, fmt));
-                    stagedRow.setLatitude(_getCellValue(row.getCell(headerMap.get("Latitude")), eval, fmt));
-                    stagedRow.setLongitude(_getCellValue(row.getCell(headerMap.get("Longitude")), eval, fmt));
-                    stagedRow.setTime(_getCellValue(row.getCell(headerMap.get("Time")), eval, fmt));
-                    stagedRow.setDate(_getCellValue(row.getCell(headerMap.get("Date")), eval, fmt));
-                    stagedRow.setVis((_getCellValue(row.getCell(headerMap.get("vis")), eval, fmt)));
-                    stagedRow.setDirection(_getCellValue(row.getCell(headerMap.get("Direction")), eval, fmt));
-                    stagedRow.setPqs(_getCellValue(row.getCell(headerMap.get("P-Qs")), eval, fmt));
-                    stagedRow.setDepth(_getCellValue(row.getCell(headerMap.get("Depth")), eval, fmt));
-                    stagedRow.setMethod(_getCellValue(row.getCell(headerMap.get("Method")), eval, fmt));
-                    stagedRow.setBlock((_getCellValue(row.getCell(headerMap.get("Block")), eval, fmt)));
-                    stagedRow.setCode(_getCellValue(row.getCell(headerMap.get("Code")), eval, fmt));
-                    stagedRow.setSpecies(_getCellValue(row.getCell(headerMap.get("Species")), eval, fmt));
-                    stagedRow.setCommonName(_getCellValue(row.getCell(headerMap.get("Common name")), eval, fmt));
-                    stagedRow.setTotal(_emptyNullTo0(_getCellValue(row.getCell(headerMap.get("Total")), eval, fmt)));
-                    stagedRow.setPos(index - 1);
-                    if (dataSheet.getHeader().size() == longHeadersRef.size()) {
-                        stagedRow.setM2InvertSizingSpecies(_getCellValue(row.getCell(headerMap.get("M2 Invert Sizing Species")), eval, fmt));
-                        stagedRow.setL5(_getCellValue(row.getCell(headerMap.get("L5")), eval, fmt));
-                        stagedRow.setL95(_getCellValue(row.getCell(headerMap.get("L95")), eval, fmt));
-                        stagedRow.setIsInvertSizing(_getCellValue(row.getCell(headerMap.get("Use InvertSizing")), eval, fmt));
-                        stagedRow.setLMax(_getCellValue(row.getCell(headerMap.get("Lmax")), eval, fmt));
+            Iterator<InputStream> sheets = xssfReader.getSheetsData();
+            while ((sheets instanceof SheetIterator) && sheets.hasNext()) {
+                InputStream i = sheets.next();
```

Comment:
This is OK, but would be more consistent with other usages if it was put into a try with resources block